python-package/lightgbm/basic.py

```diff
@@ -591,6 +600,77 @@ def _load_pandas_categorical(file_name=None, model_str=None):
         return None
 
 
+class Sequence:
+    """
+    Generic data access interface.
+
+    Object should support the following operations::
+
+        # Get total row number.
+        >>> len(seq)
+        # Random access by row index. Use for data sampling.
+        >>> seq[10]
+        # Range data access. Use to read data in batch when constructing Dataset.
+        >>> seq[0:100]
+        # Optionally specify batch_size to control range data read size.
+        >>> seq.batch_size
+
+    - With random access, **data sampling does not need to go through all data**.
+    - With range data access, there's **no need to read all data into memory thus reduce memory usage**.
```

Comment:
Document `batch_size`.
```suggestion
    - With range data access, there's **no need to read all data into memory thus reduce memory usage**.

    Attributes
    ----------
    batch_size : int
        Default size of a batch.
```